### PR TITLE
SmartAdLine Bid Adapter : initial release

### PR DESCRIPTION
**Pull request by Robot72**
## Type of change

- [x] New bidder adapter  

## Added new bid adapter

- contact email of the adapter’s maintainer
- core logic for the Smartadline Bid Adapter
- test parameters for validating bids:
```
{
  bidder: 'smartadline',
  params: {
    'publisherId': '1234'
  }
}
```

### CODE REVIEW

**Review by ChrisHuie — COMMENTED** [review]
Can you please add support for adomain and schain. Otherwise this looks good

File: modules/smartadlineBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+const BIDDER_CODE = 'smartadline';
+const ENDPOINT_URL = 'https://smartadline.com/hb.php';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+  aliases: [{code: 'smartadline'}],
```
Comment (patmmccann) [inline modules/smartadlineBidAdapter.js:9]: why is your alias the same as the bidder code

File: modules/smartadlineBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          },
+          tagid: placementCode,
+          bidfloor: bidfloor,
+          bidfloorcur: bidfloorcur
+        }],
+        site: {
+          publisher: {
+            id: publisherId
+          },
+          page: window.location.href
```
Comment (patmmccann) [inline modules/smartadlineBidAdapter.js:52]: use site.page

File: modules/smartadlineBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    if (typeof gdprConsent.gdprApplies === 'boolean') {
+      gdprParams = `gdpr=${Number(gdprConsent.gdprApplies)}&gdpr_consent=${gdprConsent.consentString}`;
+    } else {
+      gdprParams = `gdpr_consent=${gdprConsent.consentString}`;
+    }
+
+    if (syncOptions.iframeEnabled) {
+      syncs.push({
+        type: 'iframe',
+        url: '//acdn.adnxs.com/ib/static/usersync/v3/async_usersync.html?' + gdprParams
```
Comment (patmmccann) [inline modules/smartadlineBidAdapter.js:129]: always use secure

File: modules/smartadlineBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      var bid = validBidRequests[i];
+      var params = bid.params;
+      var bidId = bid.bidId;
+      var placementCode = bid.placementCode;
+      var publisherId = params.publisherId;
+      var bidfloor = params.bidfloor;
+      var bidfloorcur = params.bidfloorcur;
+      var bidRequest = {
+        id: bidId,
+        auctionId: bid.auctionId,
```
Comment (patmmccann) [inline modules/smartadlineBidAdapter.js:37]: this might be null in prebid 8

**Review by patmmccann — CHANGES_REQUESTED** [review]
Thanks for the submission!

**Comment by patmmccann** [comment]
Clearing assignee due to nonresponsive op

**Comment by patmmccann** [comment]
Stale

**Pull request closed without merging**